dynamiq/memory/backend/qdrant.py

```diff
@@ -21,122 +22,167 @@ class Qdrant(MemoryBackend):
 
     name = "Qdrant"
 
-    def __init__(self, connection: QdrantConnection, embedder: BaseEmbedder, index_name: str = "conversations"):
-        """Initializes the Qdrant memory storage."""
+    def __init__(
+        self,
+        connection: QdrantConnection,
+        embedder: BaseEmbedder,
+        index_name: str = "conversations",
+        metric: str = "cosine",
+        on_disk: bool = False,
+    ):
+        """Initializes the Qdrant memory storage.
+
+        Args:
+            connection: QdrantConnection instance for connecting to Qdrant
+            embedder: Embedder instance for creating embeddings
+            index_name: Name of the collection to store messages
+        """
         self.connection = connection
         self.index_name = index_name
         self.embedder = embedder
 
         try:
-            self.client = self.connection.connect()
+            self.vector_store = QdrantVectorStore(
+                connection=connection,
+                index_name=index_name,
+                dimension=embedder.dimensions,
+                create_if_not_exist=True,
```

Comment:
I would include these parameters at the initialization level to allow the user to decide whether to create the index or not, instead of hard-coding it to always allow index creation
```suggestion
                create_if_not_exist=create_if_not_exist,
```